src/accelerate/accelerator.py

```diff
@@ -591,6 +597,14 @@ def load_state(self, input_dir: str):
             raise ValueError(f"Tried to find {input_dir} but folder does not exist")
         logger.info(f"Loading states from {input_dir}")
         load_accelerator_state(input_dir, self._models, self._optimizers, self.state.process_index, self.scaler)
+        custom_checkpoints = [f for f in os.listdir(input_dir) if "custom_checkpoint" in f]
+        if len(custom_checkpoints) != (len(self._custom_objects) + 1):
```

Comment:
Why the +1 here?

Reply:
For the custom objects + the saved objected was my thought at the time. But you're write its a bug